### ISSUE TITLE
facilitator: plumb trace ID into all log messages

### ISSUE THREAD

**Issue by tgeoghegan** [abetterinternet/prio-server#546]
We generate a trace ID for each task scheduled by `workflow-manager` and transmit it to `facilitator` through a message queue. Some `facilitator` log messages then log the trace ID, but we should plumb it through the entire program so that all log messages emitted by `facilitator` are annotated with the trace ID to allow for more powerful log searches.

**Comment by tgeoghegan** [comment]
I'm making some good progress on this. I've got `slog-rs`, `slog-json` and `slog-term` hooked up. Example JSON log output:

```
{
    "message": "done",
    "severity": "INFO",
    "time": "2021-04-13T14:26:22.196132671-07:00",
    "facilitator_output_path": "/tmp/facil-out",
    "pha_output_path": "/tmp/pha-out",
    "batch_date": "2021/04/13/21/26",
    "batch_id": "42c040d1-b1e8-42c4-bd0e-cc5971e4afa9",
    "trace_id": "0c64c8c3-69af-43b5-9326-abe696650c32",
    "aggregation_name": "test-aggregation",
    "thread_id": "ThreadId(2)",
    "module_path": "facilitator::sample",
    "line": 362,
    "file": "src/sample.rs",
    "version": "(BUILD_INFO unavailable)"
}
```
Or, if `stderr` is a tty, you get:

![Screenshot-20210413143705-1842x153](https://user-images.githubusercontent.com/19212995/114624547-bbb89c00-9c65-11eb-9fe0-dd0d03bfdfe3.png)

I still need to figure out how to handle errors (which at a minimum need the trace ID on them), and how to carve the 3,000+ line change into multiple PRs.

**Comment by tgeoghegan** [comment]
This is going to require a few PRs, because I don't want to overwhelm my gracious and patient reviewers with a massive change. 

1. I needed to refactor sample generation to be oriented around a struct, so that we would eventually have somewhere to stick a `slog::Logger`: #559.
1. I learned that generating sample data at the command line got broken somewhere along the way, so I had to make some fixes to that: #564 
1. Once those land, we can start on the actual change to introduce `slog`. I'm going to do this a module at a time, again out of consideration for the reviewers. In #566, we introduce the `slog`, `slog_json` and `slog_term` based logging support, and convert the `aggregation` module to using a `slog::Logger` to emit structured logs. The remainder of `prio-server` uses `slog_scope` as a shim and will be converted to `slog::Logger` piecemeal in subsequent PRs. You can see what all that will look like in [this branch](https://github.com/abetterinternet/prio-server/tree/timg/slog-eventually).
